scheduler/reconcile_util_test.go

```diff
@@ -2111,18 +2123,44 @@ func TestAllocSet_filterByRescheduleable(t *testing.T) {
 					TaskGroup:    "noRescheduleTG",
 				},
 			},
-			resNow:   allocSet{},
-			resLater: []*delayedRescheduleInfo{},
+			resNow:        allocSet{},
+			resLater:      []*delayedRescheduleInfo{},
+			informational: []*structs.Allocation{},
+		},
+		{
+			name:            "allocation with sticky volumes",
+			isDisconnecting: false,
+			isBatch:         false,
+			all: allocSet{
+				"sticky": {
+					ID:            "sticky",
+					DesiredStatus: structs.AllocDesiredStatusStop,
+					ClientStatus:  structs.AllocClientStatusComplete,
+					Job:           stickyDeploymentJob,
+					HostVolumeIDs: []string{"hostVolumeID1"},
+				},
+			},
+			untainted: allocSet{},
+			resNow:    allocSet{},
+			resLater:  []*delayedRescheduleInfo{},
+			informational: []*structs.Allocation{{
+				ID:            "sticky",
+				DesiredStatus: structs.AllocDesiredStatusStop,
+				ClientStatus:  structs.AllocClientStatusComplete,
+				Job:           stickyDeploymentJob,
+				HostVolumeIDs: []string{"hostVolumeID1"},
+			}},
```

Comment:
I think we probably want to have a richer test case where we have to deal with the interactions between this feature and resheduling (addressing the comment I made on the implementation).

Reply:
Absolutely, I just adjusted this test to avoid errors.